dockerfiles/buildall-common.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+SCRIPT_DIR=$(dirname $BASH_SOURCE)
+
+##############################################################################
+#
+#    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+
+# Build script to create and publish Docker containers corresponding to a GitHub tag
+# This script is normally invoked via GitHub actions, whenever a new tag is pushed
+
+BASE_VER=7.12                                   # The docker hub label for the platform-build-base image. Changes rarely.
+BUILD_TAG=$(git describe --exact-match --tags)  # The git tag for the images we are building
+BUILD_LABEL=${BUILD_TAG}                        # The docker hub label for all other components
+BUILD_USER=hpcc-systems                         # The github repo owner
+BUILD_TYPE=                                     # Set to Debug for a debug build, leave blank for default (RelWithDebInfo)
+DOCKER_REPO=hpccsystems
+USE_CPPUNIT=1
+
+#BUILD_ML=all #ml,gnn,gnn-gpu
+ml_features=(
+  'ml'
+  'gnn'
+  'gnn-gpu'
+)
+
+# These values are set in a GitHub workflow build
+
+[[ -n ${INPUT_BUILD_USER} ]] && BUILD_USER=${INPUT_BUILD_USER}
+[[ -n ${INPUT_BUILD_VER} ]] && BUILD_TAG=${INPUT_BUILD_VER}
+[[ -n ${INPUT_DOCKER_REPO} ]] && DOCKER_REPO=${INPUT_DOCKER_REPO}
+
+if [[ -n ${INPUT_BUILD_THREADS} ]] ; then
```

Comment:
I don't see a corresponding entry in actions.yml for this variable. Is that going to be an issue if we ever try to pass it in through the github action?

Reply:
not new (was in buildall.sh) before, but you're right it's not in actions.yml.
I'm not sure we want one though, it defaults to number of detected cores, which is what you prob. want in an action.
I added this (some time ago) to override number concurrent threads used in make locally when building, not to overload my machine.